homeassistant/components/playstation_network/media_player.py

```diff
@@ -0,0 +1,139 @@
+    MediaPlayerDeviceClass,
+    MediaPlayerEntity,
+    MediaPlayerState,
+    MediaType,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import PlaystationNetworkCoordinator
+from .coordinator import PlaystationNetworkData
+from .entity import PlaystationNetworkEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@dataclass
+class PSNdata:
+    """PSN dataclass."""
+
+    account = {"id": "", "handle": ""}
+    presence = {"availability": "", "lastAvailableDate": ""}
+    platform = {"status": "", "platform": ""}
+    title = {"name": "", "format": "", "imageURL": None, "playing": False}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Media Player Entity Setup."""
+    coordinator: PlaystationNetworkCoordinator = config_entry.runtime_data
+
+    if coordinator.data.platform.get("platform") is None:
+        username = coordinator.data.username
+        _LOGGER.warning(
+            "No console found associated with account: %s. -- Skipping creation of media player",
+            username,
+        )
+        return
+
+    async_add_entities([MediaPlayer(coordinator)])
+
+
+class MediaPlayer(PlaystationNetworkEntity, MediaPlayerEntity):
+    """Media player entity representing currently playing game."""
+
+    device_class = MediaPlayerDeviceClass.TV
+
+    def __init__(self, coordinator: PlaystationNetworkCoordinator) -> None:
+        """Initialize PSN MediaPlayer."""
+        super().__init__(coordinator)
+        self.psn: PlaystationNetworkData = self.coordinator.data
+        self._attr_has_entity_name = True
+
+    @property
+    def icon(self) -> str:
+        """Icon Getter."""
+        return "mdi:sony-playstation"
+
+    @property
+    def media_image_remotely_accessible(self) -> bool:
+        """Is media image remotely accessible getter."""
+        return True
+
+    @property
+    def state(self) -> MediaPlayerState:
+        """Media Player state getter."""
+        match self.psn.platform.get("onlineStatus", ""):
+            case "online":
+                if (
+                    self.psn.available is True
+                    and self.psn.title_metadata.get("npTitleId") is not None
+                ):
+                    return MediaPlayerState.PLAYING
+                return MediaPlayerState.ON
+            case "offline":
+                return MediaPlayerState.OFF
+            case _:
+                return MediaPlayerState.OFF
+
+    @property
+    def unique_id(self) -> str:
+        """Unique ID Getter."""
+        return f"{self.psn.username.lower()}_{self.psn.platform.get('platform', "").lower()}_console"
+
+    @property
+    def name(self) -> str:
+        """Name getter."""
+        return f"{self.psn.platform.get('platform')} Console"
```

Comment:
I think you should consider not linking this to a device. Or maybe create a device for each platform the user has if that information is available in the api response?

Reply:
I know what you mean. Having to deal with multiple entities for different consoles seems annoying too. Should it just be a generic name? PlayStation Console. Something like that? 

I will take a look and confirm but I do think all of your registered consoles are available in the response. As an option.

Reply:
The XBOX integration creates one device with one media player entity per console. And another device for the account, I think that would be were the trophy sensor entities would go. We should get this done right from the beginning because changing something after it has been release is a PITA, because things shouldn't break and deprecating something is a process of min. 6 months grace period.